src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -48,10 +47,16 @@ static xccl_status_t xccl_mhba_node_fanin(xccl_mhba_team_t     *team,
             ctrl_v = (int *)((ptrdiff_t)team->node.ops[index].ctrl +
                              MHBA_CTRL_SIZE * i);
             if (*ctrl_v != request->seq_num) {
-                return XCCL_INPROGRESS;
+                if (*ctrl_v == -request->seq_num){
+                    request->buff_change_flag = 1;
+                }
+                else {
+                    return XCCL_INPROGRESS;
+                }
             }
         }
     }
+    request->buff_change_flag = request->self_change_flag ? 1 : request->buff_change_flag;
```

Comment:
This line (59) in only relevant for ASR, right? If so, i would move it into asr conditional branch, otherwise when reading the code i have to think why non_asr rank sets this (and it may not be initialized for non-asr)